### PR TITLE
Feature/backend profile

### PR DESCRIPTION
**Pull request by YevheniiBieliaiev**

### CODE REVIEW

**Review by Nightlngale — CHANGES_REQUESTED** [review]

File: packages/backend/src/api/routes/profile.ts [hunk 1]
```diff
@@ -14,9 +15,9 @@ export const initProfileRoutes = (
 
   router.get(
     apiPath(path, ProfileApiRoutes.GET_USER_BY_ID),
-    wrap((req: Request) => {
+    wrap(async (req: Request) => {
       const { userId } = req.params;
-      const user = profileService.getUser({ userId });
+      const user = profileService.getUser({ userId, req });
```
Comment (Vitalii4as) [inline packages/backend/src/api/routes/profile.ts:20]: Why req is in service

File: packages/backend/src/api/routes/profile.ts [hunk 2]
```diff
@@ -26,30 +27,38 @@ export const initProfileRoutes = (
     }),
   );
 
+  router.get(
+    apiPath(path, ProfileApiRoutes.GET_FULL_USER_DATA),
+    wrap(async (req: Request) => {
+      const { userId } = req.params;
```
Comment (Vitalii4as) [inline packages/backend/src/api/routes/profile.ts:33]: SECURITY!!! Get userId from token. This method is allowed only for authenticated user

**Review by Vitalii4as — CHANGES_REQUESTED** [review]

File: packages/backend/src/api/routes/profile.ts [hunk 3]
```diff
@@ -26,30 +27,38 @@ export const initProfileRoutes = (
...
+      const fullUserProfile = profileService.getFullUserData({ userId, req });
```
[the remark this answers is not in the document]
Comment (YevheniiBieliaiev) [reply packages/backend/src/api/routes/profile.ts:34]: My bad, it should be taken from req.userId, which is created in authMiddleware

File: packages/backend/src/api/routes/profile.ts (hunk 2 above)
Comment (YevheniiBieliaiev) [reply packages/backend/src/api/routes/profile.ts:33]: yup, my bad

File: packages/backend/src/api/routes/profile.ts (hunk 1 above)
Comment (YevheniiBieliaiev) [reply packages/backend/src/api/routes/profile.ts:20]: I didn't create a new approach. I just used what i saw in anothe routes, which was created before my task. I passed this parametr for translation of errors

Comment (YevheniiBieliaiev) [reply packages/backend/src/api/routes/profile.ts:20]: req.t('some text of error')

**Review by Nightlngale — APPROVED** [review]

**Review by Vitalii4as — APPROVED** [review]

**Pull request merged**